jobserver/models.py

```diff
@@ -108,44 +124,54 @@ def get_absolute_url(self):
         return reverse("job-detail", kwargs={"pk": self.pk})
 
     @property
-    def is_complete(self):
-        return self.status_code is None and self.completed_at
+    def is_completed(self):
+        if not self.is_finished:
+            return False
 
-    @property
-    def is_dependency_failed(self):
-        return self.status_code == 7
+        return self.status_code is None and self.completed_at
 
     @property
     def is_failed(self):
+        if not self.is_finished:
+            return False
+
+        if self.status_code is None:
+            return False
+
         dependency_not_finished = 6
         dependency_failed = 7
         dependency_running = 8
 
-        non_failure_status = self.status_code in [
+        non_failure_statuses = [
             dependency_not_finished,
             dependency_failed,
             dependency_running,
         ]
 
-        return self.status_code and not non_failure_status
+        return self.status_code not in non_failure_statuses
 
     @property
-    def is_in_progress(self):
-        if self.status_code is not None:
-            return False
-
-        return self.started_at and not self.completed_at
+    def is_finished(self):
+        return self.started and self.completed_at is not None
 
     @property
     def is_pending(self):
-        if self.status_code in [6, 8]:
+        if not self.started:
             return True
 
-        if self.status_code is None and not (self.started_at and self.completed_at):
+        if self.status_code in [6, 8]:
```

Comment:
can we use tokens again instead of integers? i.e. move the variables to module scope?

Reply:
Good point, I'll add them with a comment so we (hopefully!) remember to remove them once job-runner is driving this data.